<a href="https://colab.research.google.com/github/Sergius-DS/elecciOn_datasets/blob/main/data_diabetes_Reducido.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd
#import matplotlib.pyplot as plt
#import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format', '{:.2f}'.format)

from sklearn.model_selection import train_test_split

In [2]:
data = pd.read_csv('Afiliados_activos_DM_SIS.csv')

In [3]:
data.head()


FECHA_CORTE  FECHA_AFILIADOS_SIS  \
0     20240426             20230331   
1     20240426             20230331   
2     20240426             20230331   
3     20240426             20230331   
4     20240426             20230331   

                                  CODIGO_ANONIMIZADO  EDAD  UBIGEO  \
0  52BAEBEF1D09B9768E7B4CF5B3513538B2C45E06EF861F...    65   70106   
1  7FD6261B4EA46FF607C1BA139B94C73CC0446E2FC7B877...    13   80907   
2  FFBE798BE21A94EB56BB8A0B7FBDEC07AFB8824A688294...    96  150701   
3  A063A90D386099DDBD7AEA15F5F2A96FDCA3B2E326A02C...    60   20801   
4  648654B6F9D7538441DAA9987966AD163941F0755C809B...    56   70101   

  DEPARTAMENTO                PROVINCIA    DISTRITO       SEXO  \
0       CALLAO  PROV. CONST. DEL CALLAO  VENTANILLA   FEMENINO   
1        CUSCO            LA CONVENCIÓN     KIMBIRI  MASCULINO   
2         LIMA               HUAROCHIRÍ    MATUCANA   FEMENINO   
3       ÁNCASH                    CASMA       CASMA   FEMENINO   
4       CALLAO  PROV. CONST. DEL CALLAO      CALLAO  MASCULINO   

   FECHA_PRIMER_DX                       TIPO_DIABETES CON_DX_OBESIDAD  \
0         20220620  Diabetes mellitus  no especificada             NaN   
1         20221001            Diabetes mellitus tipo 2             NaN   
2         20191216            Diabetes mellitus tipo 2              SI   
3         20191216            Diabetes mellitus tipo 1             NaN   
4         20181029  Diabetes mellitus  no especificada              SI   

  CON_DX_HIPERTENSION CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  \
0                 NaN                NaN                0        0.00   
1                 NaN                NaN                0        0.00   
2                 NaN                 SI                0        0.00   
3                  SI                NaN                0        0.00   
4                  SI                NaN                1       17.73   

   CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  DIAS_HOSP  UBIGEO_ULT_ATE  \
0                     0             0.00          0             NaN   
1                     0             0.00          0             NaN   
2                     0             0.00          0             NaN   
3                     0             0.00          0             NaN   
4                     0             0.00          0        70101.00   

  DEPARTAMENTO_ULT_ATE        PROVINCIA_ULT_ATE DISTRITO_ULT_ATE  \
0                  NaN                      NaN              NaN   
1                  NaN                      NaN              NaN   
2                  NaN                      NaN              NaN   
3                  NaN                      NaN              NaN   
4               CALLAO  PROV. CONST. DEL CALLAO           CALLAO   

   NIVEL_ULT_ATE  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4           1.00

In [4]:
data.shape

(1735940, 24)

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1735940 entries, 0 to 1735939
Data columns (total 24 columns):
 #   Column                Dtype  
---  ------                -----  
 0   FECHA_CORTE           int64  
 1   FECHA_AFILIADOS_SIS   int64  
 2   CODIGO_ANONIMIZADO    object 
 3   EDAD                  int64  
 4   UBIGEO                int64  
 5   DEPARTAMENTO          object 
 6   PROVINCIA             object 
 7   DISTRITO              object 
 8   SEXO                  object 
 9   FECHA_PRIMER_DX       int64  
 10  TIPO_DIABETES         object 
 11  CON_DX_OBESIDAD       object 
 12  CON_DX_HIPERTENSION   object 
 13  CON_DX_SALUDMENTAL    object 
 14  CANT_ATENCIONES       int64  
 15  VALOR_NETO            float64
 16  CANT_ATENCIONES_HOSP  int64  
 17  VALOR_NETO_HOSP       float64
 18  DIAS_HOSP             int64  
 19  UBIGEO_ULT_ATE        float64
 20  DEPARTAMENTO_ULT_ATE  object 
 21  PROVINCIA_ULT_ATE     object 
 22  DISTRITO_ULT_ATE      object 
 23  NIVEL_U

In [6]:
data.FECHA_CORTE.value_counts()

FECHA_CORTE
20240426    1735940
Name: count, dtype: int64

In [7]:
data.drop('FECHA_CORTE', axis=1, inplace=True)

In [8]:
data.DEPARTAMENTO.value_counts()

DEPARTAMENTO
LIMA             668296
PIURA            134433
LA LIBERTAD      116812
CALLAO            89114
SAN MARTÍN        76216
LORETO            68298
ÁNCASH            66793
LAMBAYEQUE        65688
ICA               52180
AREQUIPA          49804
CAJAMARCA         48102
JUNÍN             45480
UCAYALI           35020
CUSCO             33150
HUÁNUCO           27102
TUMBES            25163
PUNO              22772
AYACUCHO          21604
TACNA             21353
AMAZONAS          16648
MADRE DE DIOS     13469
APURÍMAC          11308
MOQUEGUA          10767
PASCO              8334
HUANCAVELICA       8034
Name: count, dtype: int64

In [9]:
data.DIAS_HOSP.value_counts()

DIAS_HOSP
0      1709458
3         2266
4         2136
2         1977
5         1929
        ...   
127          1
198          1
172          1
125          1
138          1
Name: count, Length: 161, dtype: int64

In [10]:
# Creando  la nueva columna HOSPITALIZADO
data['HOSPITALIZADO'] = data['DIAS_HOSP'].apply(lambda x: 1 if x > 0 else 0)

In [11]:
data.HOSPITALIZADO.value_counts()

HOSPITALIZADO
0    1709458
1      26482
Name: count, dtype: int64

In [12]:
data.isnull().sum()

FECHA_AFILIADOS_SIS           0
CODIGO_ANONIMIZADO            0
EDAD                          0
UBIGEO                        0
DEPARTAMENTO                  0
PROVINCIA                     0
DISTRITO                      0
SEXO                          0
FECHA_PRIMER_DX               0
TIPO_DIABETES                 0
CON_DX_OBESIDAD          633509
CON_DX_HIPERTENSION     1080678
CON_DX_SALUDMENTAL      1357231
CANT_ATENCIONES               0
VALOR_NETO                    0
CANT_ATENCIONES_HOSP          0
VALOR_NETO_HOSP               0
DIAS_HOSP                     0
UBIGEO_ULT_ATE          1214440
DEPARTAMENTO_ULT_ATE    1214440
PROVINCIA_ULT_ATE       1214440
DISTRITO_ULT_ATE        1214440
NIVEL_ULT_ATE           1214444
HOSPITALIZADO                 0
dtype: int64

In [13]:
# Porcentaje Nulo por columnas
missing_percentage = (data.isnull().sum() / len(data)) * 100
missing_percentage

FECHA_AFILIADOS_SIS     0.00
CODIGO_ANONIMIZADO      0.00
EDAD                    0.00
UBIGEO                  0.00
DEPARTAMENTO            0.00
PROVINCIA               0.00
DISTRITO                0.00
SEXO                    0.00
FECHA_PRIMER_DX         0.00
TIPO_DIABETES           0.00
CON_DX_OBESIDAD        36.49
CON_DX_HIPERTENSION    62.25
CON_DX_SALUDMENTAL     78.18
CANT_ATENCIONES         0.00
VALOR_NETO              0.00
CANT_ATENCIONES_HOSP    0.00
VALOR_NETO_HOSP         0.00
DIAS_HOSP               0.00
UBIGEO_ULT_ATE         69.96
DEPARTAMENTO_ULT_ATE   69.96
PROVINCIA_ULT_ATE      69.96
DISTRITO_ULT_ATE       69.96
NIVEL_ULT_ATE          69.96
HOSPITALIZADO           0.00
dtype: float64

In [14]:
data.duplicated().sum()

0

In [15]:
data.CON_DX_HIPERTENSION.value_counts()

CON_DX_HIPERTENSION
SI    655262
Name: count, dtype: int64

In [16]:
data.CON_DX_OBESIDAD.value_counts()

CON_DX_OBESIDAD
SI    1102431
Name: count, dtype: int64

In [17]:
# Lista de columnas a transformar
columns_to_transform = ['CON_DX_OBESIDAD', 'CON_DX_HIPERTENSION', 'CON_DX_SALUDMENTAL']

# Reemplazar valores y convertir a entero
for column in columns_to_transform:
    data[column] = data[column].apply(lambda x: 1 if x == 'SI' else 0).astype(int)

# Verifica las primeras filas para confirmar la transformación
print(data[columns_to_transform].head())

   CON_DX_OBESIDAD  CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL
0                0                    0                   0
1                0                    0                   0
2                1                    0                   1
3                0                    1                   0
4                1                    1                   0


In [18]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1735940 entries, 0 to 1735939
Data columns (total 24 columns):
 #   Column                Dtype  
---  ------                -----  
 0   FECHA_AFILIADOS_SIS   int64  
 1   CODIGO_ANONIMIZADO    object 
 2   EDAD                  int64  
 3   UBIGEO                int64  
 4   DEPARTAMENTO          object 
 5   PROVINCIA             object 
 6   DISTRITO              object 
 7   SEXO                  object 
 8   FECHA_PRIMER_DX       int64  
 9   TIPO_DIABETES         object 
 10  CON_DX_OBESIDAD       int64  
 11  CON_DX_HIPERTENSION   int64  
 12  CON_DX_SALUDMENTAL    int64  
 13  CANT_ATENCIONES       int64  
 14  VALOR_NETO            float64
 15  CANT_ATENCIONES_HOSP  int64  
 16  VALOR_NETO_HOSP       float64
 17  DIAS_HOSP             int64  
 18  UBIGEO_ULT_ATE        float64
 19  DEPARTAMENTO_ULT_ATE  object 
 20  PROVINCIA_ULT_ATE     object 
 21  DISTRITO_ULT_ATE      object 
 22  NIVEL_ULT_ATE         float64
 23  HOSPITA

In [19]:
data.isna().sum()

FECHA_AFILIADOS_SIS           0
CODIGO_ANONIMIZADO            0
EDAD                          0
UBIGEO                        0
DEPARTAMENTO                  0
PROVINCIA                     0
DISTRITO                      0
SEXO                          0
FECHA_PRIMER_DX               0
TIPO_DIABETES                 0
CON_DX_OBESIDAD               0
CON_DX_HIPERTENSION           0
CON_DX_SALUDMENTAL            0
CANT_ATENCIONES               0
VALOR_NETO                    0
CANT_ATENCIONES_HOSP          0
VALOR_NETO_HOSP               0
DIAS_HOSP                     0
UBIGEO_ULT_ATE          1214440
DEPARTAMENTO_ULT_ATE    1214440
PROVINCIA_ULT_ATE       1214440
DISTRITO_ULT_ATE        1214440
NIVEL_ULT_ATE           1214444
HOSPITALIZADO                 0
dtype: int64

In [20]:
data.UBIGEO_ULT_ATE.value_counts()

UBIGEO_ULT_ATE
150132.00    19316
150101.00    18549
150135.00    14012
150103.00    13963
150133.00    12881
             ...  
101007.00        1
90402.00         1
40512.00         1
30713.00         1
160702.00        1
Name: count, Length: 1774, dtype: int64

In [21]:
# Eliminamos las columnas referidas
data.drop(['UBIGEO','UBIGEO_ULT_ATE', 'DEPARTAMENTO_ULT_ATE', 'PROVINCIA_ULT_ATE', 'DISTRITO_ULT_ATE', 'NIVEL_ULT_ATE'], axis=1, inplace=True)

In [22]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1735940 entries, 0 to 1735939
Data columns (total 18 columns):
 #   Column                Dtype  
---  ------                -----  
 0   FECHA_AFILIADOS_SIS   int64  
 1   CODIGO_ANONIMIZADO    object 
 2   EDAD                  int64  
 3   DEPARTAMENTO          object 
 4   PROVINCIA             object 
 5   DISTRITO              object 
 6   SEXO                  object 
 7   FECHA_PRIMER_DX       int64  
 8   TIPO_DIABETES         object 
 9   CON_DX_OBESIDAD       int64  
 10  CON_DX_HIPERTENSION   int64  
 11  CON_DX_SALUDMENTAL    int64  
 12  CANT_ATENCIONES       int64  
 13  VALOR_NETO            float64
 14  CANT_ATENCIONES_HOSP  int64  
 15  VALOR_NETO_HOSP       float64
 16  DIAS_HOSP             int64  
 17  HOSPITALIZADO         int64  
dtypes: float64(2), int64(10), object(6)
memory usage: 238.4+ MB


In [23]:
data.describe()

FECHA_AFILIADOS_SIS       EDAD  FECHA_PRIMER_DX  CON_DX_OBESIDAD  \
count           1735940.00 1735940.00       1735940.00       1735940.00   
mean           20230640.14      55.75      20203280.89             0.64   
std                 244.34      16.81         17695.02             0.48   
min            20230331.00       0.00      20180101.00             0.00   
25%            20230331.00      46.00      20190122.00             0.00   
50%            20230630.00      57.00      20201019.00             1.00   
75%            20230930.00      68.00      20220523.00             1.00   
max            20230930.00     119.00      20230930.00             1.00   

       CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  \
count           1735940.00          1735940.00       1735940.00  1735940.00   
mean                  0.38                0.22             0.56       52.88   
std                   0.48                0.41             1.15      821.75   
min                   0.00                0.00             0.00        0.00   
25%                   0.00                0.00             0.00        0.00   
50%                   0.00                0.00             0.00        0.00   
75%                   1.00                0.00             1.00        0.00   
max                   1.00                1.00           134.00   232695.27   

       CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
count            1735940.00       1735940.00 1735940.00     1735940.00  
mean                   0.02            40.63       0.18           0.02  
std                    0.15           811.69       2.58           0.12  
min                    0.00             0.00       0.00           0.00  
25%                    0.00             0.00       0.00           0.00  
50%                    0.00             0.00       0.00           0.00  
75%                    0.00             0.00       0.00           0.00  
max                   13.00        232695.27     378.00           1.00

In [24]:
# Lista de columnas de fechas
date_columns = ['FECHA_AFILIADOS_SIS', 'FECHA_PRIMER_DX']

# Convertir a datetime y formatear
for col in date_columns:
    data[col] = pd.to_datetime(data[col], format='%Y%m%d', errors='coerce')  # Asumiendo formato AAAAMMDD


In [25]:
data.head()

FECHA_AFILIADOS_SIS                                 CODIGO_ANONIMIZADO  \
0          2023-03-31  52BAEBEF1D09B9768E7B4CF5B3513538B2C45E06EF861F...   
1          2023-03-31  7FD6261B4EA46FF607C1BA139B94C73CC0446E2FC7B877...   
2          2023-03-31  FFBE798BE21A94EB56BB8A0B7FBDEC07AFB8824A688294...   
3          2023-03-31  A063A90D386099DDBD7AEA15F5F2A96FDCA3B2E326A02C...   
4          2023-03-31  648654B6F9D7538441DAA9987966AD163941F0755C809B...   

   EDAD DEPARTAMENTO                PROVINCIA    DISTRITO       SEXO  \
0    65       CALLAO  PROV. CONST. DEL CALLAO  VENTANILLA   FEMENINO   
1    13        CUSCO            LA CONVENCIÓN     KIMBIRI  MASCULINO   
2    96         LIMA               HUAROCHIRÍ    MATUCANA   FEMENINO   
3    60       ÁNCASH                    CASMA       CASMA   FEMENINO   
4    56       CALLAO  PROV. CONST. DEL CALLAO      CALLAO  MASCULINO   

  FECHA_PRIMER_DX                       TIPO_DIABETES  CON_DX_OBESIDAD  \
0      2022-06-20  Diabetes mellitus  no especificada                0   
1      2022-10-01            Diabetes mellitus tipo 2                0   
2      2019-12-16            Diabetes mellitus tipo 2                1   
3      2019-12-16            Diabetes mellitus tipo 1                0   
4      2018-10-29  Diabetes mellitus  no especificada                1   

   CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  \
0                    0                   0                0        0.00   
1                    0                   0                0        0.00   
2                    0                   1                0        0.00   
3                    1                   0                0        0.00   
4                    1                   0                1       17.73   

   CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
0                     0             0.00          0              0  
1                     0             0.00          0              0  
2                     0             0.00          0              0  
3                     0             0.00          0              0  
4                     0             0.00          0              0

In [26]:
data.describe()

FECHA_AFILIADOS_SIS       EDAD  \
count                        1735940 1735940.00   
mean   2023-07-03 07:56:01.313409024      55.75   
min              2023-03-31 00:00:00       0.00   
25%              2023-03-31 00:00:00      46.00   
50%              2023-06-30 00:00:00      57.00   
75%              2023-09-30 00:00:00      68.00   
max              2023-09-30 00:00:00     119.00   
std                              NaN      16.81   

                     FECHA_PRIMER_DX  CON_DX_OBESIDAD  CON_DX_HIPERTENSION  \
count                        1735940       1735940.00           1735940.00   
mean   2020-09-19 21:01:33.047455232             0.64                 0.38   
min              2018-01-01 00:00:00             0.00                 0.00   
25%              2019-01-22 00:00:00             0.00                 0.00   
50%              2020-10-19 00:00:00             1.00                 0.00   
75%              2022-05-23 00:00:00             1.00                 1.00   
max              2023-09-30 00:00:00             1.00                 1.00   
std                              NaN             0.48                 0.48   

       CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  CANT_ATENCIONES_HOSP  \
count          1735940.00       1735940.00  1735940.00            1735940.00   
mean                 0.22             0.56       52.88                  0.02   
min                  0.00             0.00        0.00                  0.00   
25%                  0.00             0.00        0.00                  0.00   
50%                  0.00             0.00        0.00                  0.00   
75%                  0.00             1.00        0.00                  0.00   
max                  1.00           134.00   232695.27                 13.00   
std                  0.41             1.15      821.75                  0.15   

       VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
count       1735940.00 1735940.00     1735940.00  
mean             40.63       0.18           0.02  
min               0.00       0.00           0.00  
25%               0.00       0.00           0.00  
50%               0.00       0.00           0.00  
75%               0.00       0.00           0.00  
max          232695.27     378.00           1.00  
std             811.69       2.58           0.12

In [27]:
# Identificar las columnas numéricas
numeric_columns = data.select_dtypes(include=['int64', 'float64']).columns

# Filtrar columnas con valores binarios (0 y 1)
binary_columns = [
    col for col in numeric_columns if data[col].dropna().isin([0, 1]).all()
]

# Convertir las columnas binarias a int8
data[binary_columns] = data[binary_columns].astype('int8')

# Verificar el cambio de tipo
print(data[binary_columns].dtypes)


CON_DX_OBESIDAD        int8
CON_DX_HIPERTENSION    int8
CON_DX_SALUDMENTAL     int8
HOSPITALIZADO          int8
dtype: object


In [28]:
data['EDAD'] = data['EDAD'].astype('int8')

In [29]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1735940 entries, 0 to 1735939
Data columns (total 18 columns):
 #   Column                Dtype         
---  ------                -----         
 0   FECHA_AFILIADOS_SIS   datetime64[ns]
 1   CODIGO_ANONIMIZADO    object        
 2   EDAD                  int8          
 3   DEPARTAMENTO          object        
 4   PROVINCIA             object        
 5   DISTRITO              object        
 6   SEXO                  object        
 7   FECHA_PRIMER_DX       datetime64[ns]
 8   TIPO_DIABETES         object        
 9   CON_DX_OBESIDAD       int8          
 10  CON_DX_HIPERTENSION   int8          
 11  CON_DX_SALUDMENTAL    int8          
 12  CANT_ATENCIONES       int64         
 13  VALOR_NETO            float64       
 14  CANT_ATENCIONES_HOSP  int64         
 15  VALOR_NETO_HOSP       float64       
 16  DIAS_HOSP             int64         
 17  HOSPITALIZADO         int8          
dtypes: datetime64[ns](2), float64(2), int64(3)

In [30]:
data[data['CODIGO_ANONIMIZADO'].duplicated()].count()

FECHA_AFILIADOS_SIS     1109984
CODIGO_ANONIMIZADO      1109984
EDAD                    1109984
DEPARTAMENTO            1109984
PROVINCIA               1109984
DISTRITO                1109984
SEXO                    1109984
FECHA_PRIMER_DX         1109984
TIPO_DIABETES           1109984
CON_DX_OBESIDAD         1109984
CON_DX_HIPERTENSION     1109984
CON_DX_SALUDMENTAL      1109984
CANT_ATENCIONES         1109984
VALOR_NETO              1109984
CANT_ATENCIONES_HOSP    1109984
VALOR_NETO_HOSP         1109984
DIAS_HOSP               1109984
HOSPITALIZADO           1109984
dtype: int64

In [31]:
data = data[data['FECHA_AFILIADOS_SIS'] >= '2023-09-01']

In [32]:
data[data['CODIGO_ANONIMIZADO'].duplicated()].count()

FECHA_AFILIADOS_SIS     0
CODIGO_ANONIMIZADO      0
EDAD                    0
DEPARTAMENTO            0
PROVINCIA               0
DISTRITO                0
SEXO                    0
FECHA_PRIMER_DX         0
TIPO_DIABETES           0
CON_DX_OBESIDAD         0
CON_DX_HIPERTENSION     0
CON_DX_SALUDMENTAL      0
CANT_ATENCIONES         0
VALOR_NETO              0
CANT_ATENCIONES_HOSP    0
VALOR_NETO_HOSP         0
DIAS_HOSP               0
HOSPITALIZADO           0
dtype: int64

In [33]:
data[data['CODIGO_ANONIMIZADO'] == '110D1BC55E119D0732C0B3C17A92962F4EA4C04718F08DBF632F2A1BB402BF7F']

FECHA_AFILIADOS_SIS                                 CODIGO_ANONIMIZADO  \
550183          2023-09-30  110D1BC55E119D0732C0B3C17A92962F4EA4C04718F08D...   

        EDAD DEPARTAMENTO PROVINCIA    DISTRITO       SEXO FECHA_PRIMER_DX  \
550183    55          ICA     PISCO  SAN ANDRÉS  MASCULINO      2022-03-01   

                   TIPO_DIABETES  CON_DX_OBESIDAD  CON_DX_HIPERTENSION  \
550183  Diabetes mellitus tipo 2                1                    0   

        CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  CANT_ATENCIONES_HOSP  \
550183                   0                0        0.00                     0   

        VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
550183             0.00          0              0

In [34]:
data[data['CODIGO_ANONIMIZADO'] == '02E115774EDA02CAD46ECED7D90F38398F399115B916FF14C0DA5F372C195B20']

FECHA_AFILIADOS_SIS                                 CODIGO_ANONIMIZADO  \
550331          2023-09-30  02E115774EDA02CAD46ECED7D90F38398F399115B916FF...   

        EDAD DEPARTAMENTO PROVINCIA DISTRITO      SEXO FECHA_PRIMER_DX  \
550331    72     APURÍMAC   ABANCAY  ABANCAY  FEMENINO      2022-05-03   

                   TIPO_DIABETES  CON_DX_OBESIDAD  CON_DX_HIPERTENSION  \
550331  Diabetes mellitus tipo 2                0                    0   

        CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  CANT_ATENCIONES_HOSP  \
550331                   0                0        0.00                     0   

        VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
550331             0.00          0              0

In [35]:
data[(data['EDAD'] == 0) | (data['EDAD'] <= 12)].count()

FECHA_AFILIADOS_SIS     8038
CODIGO_ANONIMIZADO      8038
EDAD                    8038
DEPARTAMENTO            8038
PROVINCIA               8038
DISTRITO                8038
SEXO                    8038
FECHA_PRIMER_DX         8038
TIPO_DIABETES           8038
CON_DX_OBESIDAD         8038
CON_DX_HIPERTENSION     8038
CON_DX_SALUDMENTAL      8038
CANT_ATENCIONES         8038
VALOR_NETO              8038
CANT_ATENCIONES_HOSP    8038
VALOR_NETO_HOSP         8038
DIAS_HOSP               8038
HOSPITALIZADO           8038
dtype: int64

In [36]:
data = data[data['EDAD'] > 11]

OBSERVACIONES:

Se filtra EDAD con criterio de valores mayores de 10 años

In [37]:
data = data[data['EDAD'] < 101]

OBSERVACIONES:

Se filtra EDAD con criterio de valores hasta 100 años

In [38]:
data['EDAD'].describe()

count   600043.00
mean        56.02
std         16.08
min         12.00
25%         46.00
50%         57.00
75%         67.00
max        100.00
Name: EDAD, dtype: float64

OBSERVACIONES:

Se obtiene en la variable EDAD un rango entre 12 y 100 años

In [39]:
data[['VALOR_NETO', 'VALOR_NETO_HOSP']].describe()

VALOR_NETO  VALOR_NETO_HOSP
count   600043.00        600043.00
mean        51.13            39.28
std        774.46           762.23
min          0.00             0.00
25%          0.00             0.00
50%          0.00             0.00
75%          0.00             0.00
max     169866.74        169866.74

OBSERVACIONES:

Datos estadísticos de las variables VALOR_NETO y	VALOR_NETO_HOSP

In [40]:
#quantile_75_valor_neto = data['VALOR_NETO'].quantile(0.75)

In [41]:
quantile_95_valor_neto = data['VALOR_NETO'].quantile(0.95)

In [42]:
mayores_valores_neto = data[data['VALOR_NETO'] > quantile_95_valor_neto]['VALOR_NETO'].sort_values(ascending=False)
mayores_valores_neto

1581090   169866.74
1204767   135748.75
1582068   130109.48
1357314   115796.45
1539903    86568.58
             ...   
1221426       94.83
1268698       94.82
1251266       94.82
1170466       94.82
1159534       94.81
Name: VALOR_NETO, Length: 30001, dtype: float64

OBSERVACIONES:

Se visualiza en primera instancia los valores máximos, para ver su frecuencia en la columna

VISUALIZACIÓN PARCIAL COSTOS MÁS ELEVADOS

In [43]:
data.loc[mayores_valores_neto.index].head(10)

FECHA_AFILIADOS_SIS  \
1581090          2023-09-30   
1204767          2023-09-30   
1582068          2023-09-30   
1357314          2023-09-30   
1539903          2023-09-30   
1731206          2023-09-30   
1158871          2023-09-30   
1316012          2023-09-30   
1696048          2023-09-30   
1352860          2023-09-30   

                                        CODIGO_ANONIMIZADO  EDAD DEPARTAMENTO  \
1581090  F8A54B544342BAE1CF8DC52802F6FC4198133E1C6473DC...    51    CAJAMARCA   
1204767  0FDF79C7105B9BBA3C17FC44E64F1E3D2808DF477C527C...    51        PIURA   
1582068  5A7F9F8A9E4AAF59387587C49C096E696B9D13383D87C8...    63         LIMA   
1357314  845E695415F7157369D55F2F6BB6A0D1A848C9596B080E...    47        PASCO   
1539903  83EB0644C2DD0A13364DA189BA2EB9606779864AB06F5E...    65          ICA   
1731206  E74B14C305DC6F6CC93654E95F7B5EE03E997947E3302A...    37         LIMA   
1158871  4A2AAADF42FC4DF6D1B76427FC4894CAAB8F8885DDDACC...    63         LIMA   
1316012  AE615C955B7123BFE9851A1A61BCCCA3B304ED858BA7A7...    40         LIMA   
1696048  522B87D89350428F5DD7335C3A123E33E52D1304E2AB99...    23  LA LIBERTAD   
1352860  1FEEA54C1855197E6ED6F84FB263B43194A2232CFD1613...    70       ÁNCASH   

        PROVINCIA           DISTRITO       SEXO FECHA_PRIMER_DX  \
1581090     CHOTA              LLAMA   FEMENINO      2018-04-16   
1204767     PIURA       TAMBO GRANDE  MASCULINO      2023-08-21   
1582068      LIMA           BARRANCO  MASCULINO      2023-08-18   
1357314  OXAPAMPA         VILLA RICA   FEMENINO      2023-08-29   
1539903   CHINCHA       CHINCHA BAJA   FEMENINO      2023-08-26   
1731206      LIMA  SANTIAGO DE SURCO  MASCULINO      2023-04-21   
1158871      LIMA         LOS OLIVOS   FEMENINO      2019-01-23   
1316012      LIMA         LOS OLIVOS  MASCULINO      2023-08-03   
1696048      VIRÚ               CHAO  MASCULINO      2023-07-01   
1352860     HUARI  CHAVIN DE HUANTAR   FEMENINO      2020-07-31   

                                       TIPO_DIABETES  CON_DX_OBESIDAD  \
1581090           Diabetes mellitus  no especificada                1   
1204767                     Diabetes mellitus tipo 2                0   
1582068           Diabetes mellitus  no especificada                0   
1357314                     Diabetes mellitus tipo 2                0   
1539903  Diabetes mellitus asociada con desnutrición                0   
1731206                     Diabetes mellitus tipo 2                0   
1158871                     Diabetes mellitus tipo 2                1   
1316012           Diabetes mellitus  no especificada                0   
1696048                     Diabetes mellitus tipo 1                0   
1352860           Diabetes mellitus  no especificada                0   

         CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  \
1581090                    1                   0                2   169866.74   
1204767                    1                   0                1   135748.75   
1582068                    0                   0                1   130109.48   
1357314                    1                   0                1   115796.45   
1539903                    1                   1                1    86568.58   
1731206                    0                   0                1    78163.32   
1158871                    0                   1                2    64147.85   
1316012                    0                   0                1    62437.67   
1696048                    1                   1                1    62241.01   
1352860                    1                   1                1    59489.19   

         CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
1581090                     2        169866.74         59              1  
1204767                     1        135748.75         91              1  
1582068                     1        130109.48         35              1  
1357314                     1        115

OBSERVACIONES:

Se puede observar los casos más costos ligados a una sola atención con los top 3:

PRIMERO: 135748.75

SEGUNDO: 130109.48

TERCERO: 115796.45

In [44]:
mayores_valores_neto.head(5).value_counts()

VALOR_NETO
169866.74    1
135748.75    1
130109.48    1
115796.45    1
86568.58     1
Name: count, dtype: int64

OBSERVACIONES:

Se muestran los primeros 5 valores máximos de VALOR_NETO

In [45]:
data['VALOR_NETO'].max()

169866.74

OBSERVACIONES:

Hay solo una ocurrencia del valor máximo en VALOR_NETO de "169866.74", pero si dividimos la cantidad de atenciones 2, nos da que cada atención costó 84933.37 SOLES

In [46]:
data[data['VALOR_NETO'] == 169866.74]

FECHA_AFILIADOS_SIS  \
1581090          2023-09-30   

                                        CODIGO_ANONIMIZADO  EDAD DEPARTAMENTO  \
1581090  F8A54B544342BAE1CF8DC52802F6FC4198133E1C6473DC...    51    CAJAMARCA   

        PROVINCIA DISTRITO      SEXO FECHA_PRIMER_DX  \
1581090     CHOTA    LLAMA  FEMENINO      2018-04-16   

                              TIPO_DIABETES  CON_DX_OBESIDAD  \
1581090  Diabetes mellitus  no especificada                1   

         CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  \
1581090                    1                   0                2   169866.74   

         CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
1581090                     2        169866.74         59              1

OBSERVACIONES:

El VALOR_NETO máximo es por 2 atenciones

In [47]:
(data['VALOR_NETO'] == data['VALOR_NETO'].max()).sum()

1

In [48]:
data['VALOR_NETO_HOSP'].max()

169866.74

OBSERVACIONES:

Hay solo una ocurrencia del valor máximo en VALOR_NETO_HOSP

In [49]:
data[data['VALOR_NETO_HOSP'] == 169866.74]

FECHA_AFILIADOS_SIS  \
1581090          2023-09-30   

                                        CODIGO_ANONIMIZADO  EDAD DEPARTAMENTO  \
1581090  F8A54B544342BAE1CF8DC52802F6FC4198133E1C6473DC...    51    CAJAMARCA   

        PROVINCIA DISTRITO      SEXO FECHA_PRIMER_DX  \
1581090     CHOTA    LLAMA  FEMENINO      2018-04-16   

                              TIPO_DIABETES  CON_DX_OBESIDAD  \
1581090  Diabetes mellitus  no especificada                1   

         CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  \
1581090                    1                   0                2   169866.74   

         CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
1581090                     2        169866.74         59              1

In [50]:
(data['VALOR_NETO_HOSP'] == data['VALOR_NETO_HOSP'].max()).sum()

1

In [51]:
data[data['VALOR_NETO_HOSP'] > data['VALOR_NETO_HOSP'].quantile(0.95)]\
['VALOR_NETO_HOSP'].sort_values(ascending=False).head().value_counts()

VALOR_NETO_HOSP
169866.74    1
135748.75    1
130109.48    1
115796.45    1
86568.58     1
Name: count, dtype: int64

OBSERVACIONES:

Se muestran los primeros 5 valores máximos de VALOR_NETO_HOSP

In [52]:
data.shape

(600043, 18)

In [53]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 600043 entries, 550183 to 1735939
Data columns (total 18 columns):
 #   Column                Non-Null Count   Dtype         
---  ------                --------------   -----         
 0   FECHA_AFILIADOS_SIS   600043 non-null  datetime64[ns]
 1   CODIGO_ANONIMIZADO    600043 non-null  object        
 2   EDAD                  600043 non-null  int8          
 3   DEPARTAMENTO          600043 non-null  object        
 4   PROVINCIA             600043 non-null  object        
 5   DISTRITO              600043 non-null  object        
 6   SEXO                  600043 non-null  object        
 7   FECHA_PRIMER_DX       600043 non-null  datetime64[ns]
 8   TIPO_DIABETES         600043 non-null  object        
 9   CON_DX_OBESIDAD       600043 non-null  int8          
 10  CON_DX_HIPERTENSION   600043 non-null  int8          
 11  CON_DX_SALUDMENTAL    600043 non-null  int8          
 12  CANT_ATENCIONES       600043 non-null  int64         
 13

In [54]:
# Verifica las categorías en la columna objetivo
print(f"Proporción antes de Reducir Set de Datos: \n{data['HOSPITALIZADO'].value_counts(normalize=True)}")

Proporción antes de Reducir Set de Datos: 
HOSPITALIZADO
0   0.98
1   0.02
Name: proportion, dtype: float64


REDUCCIÓN DEL SET DE DATOS

In [55]:
# Realiza un muestreo estratificado para reducir a 50% del tamaño original
data_sample, _ = train_test_split(
    data,
    test_size=0.5,  # Mantén el 50% del dataset original
    stratify=data['HOSPITALIZADO'],  # Estratifica por la columna HOSPITALIZADO
    random_state=42  # Para reproducibilidad
)

In [56]:
print(f"Proporción después de Reducir Set de Datos: \n{data_sample['HOSPITALIZADO'].value_counts(normalize=True)}")

Proporción después de Reducir Set de Datos: 
HOSPITALIZADO
0   0.98
1   0.02
Name: proportion, dtype: float64


OBSERVACIONES:

Al reducir el set de datos se mantiene la misma propoción

In [57]:
# Distribución de variable Objetivo
data_sample['HOSPITALIZADO'].value_counts()

HOSPITALIZADO
0    295445
1      4576
Name: count, dtype: int64

OBSERVACIONES:

La distribución de la variable objetivo, pero con la cuenta total de las 2 categorías, NO_HOSPITALIZADO: 295445 casos y HOSPITALIZADO: 4576 casos

In [58]:
data_sample.info()

<class 'pandas.core.frame.DataFrame'>
Index: 300021 entries, 1716388 to 1709190
Data columns (total 18 columns):
 #   Column                Non-Null Count   Dtype         
---  ------                --------------   -----         
 0   FECHA_AFILIADOS_SIS   300021 non-null  datetime64[ns]
 1   CODIGO_ANONIMIZADO    300021 non-null  object        
 2   EDAD                  300021 non-null  int8          
 3   DEPARTAMENTO          300021 non-null  object        
 4   PROVINCIA             300021 non-null  object        
 5   DISTRITO              300021 non-null  object        
 6   SEXO                  300021 non-null  object        
 7   FECHA_PRIMER_DX       300021 non-null  datetime64[ns]
 8   TIPO_DIABETES         300021 non-null  object        
 9   CON_DX_OBESIDAD       300021 non-null  int8          
 10  CON_DX_HIPERTENSION   300021 non-null  int8          
 11  CON_DX_SALUDMENTAL    300021 non-null  int8          
 12  CANT_ATENCIONES       300021 non-null  int64         
 1

In [59]:
#data_sample.to_csv('data_diabetes.csv', index=False)

In [60]:
#ds = pd.read_csv('data_diabetes.csv')

In [61]:
#ds.info()